### PR TITLE
docs: Revised numaflow introduction

### PR DESCRIPTION
**Pull request by syayi**
Revised numaflow introduction

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,29 +7,27 @@
...
 [![CII Best Practices](https://bestpractices.coreinfrastructure.org/projects/6078/badge)](https://bestpractices.coreinfrastructure.org/projects/6078)
 
-## Summary
+Welcome to Numaflow, a Kubernetes-native platform designed for developing scalable and reliable event-driven applications. Numaflow decouples event sources and sinks from the processing logic, allowing each component to independently auto-scale based on demand. With out-of-the-box sources and sinks, developers can focus solely on the processing logic without worrying about event consumption, boilerplate code, or operational complexities. Each step of the pipeline can be written in any programming language, offering unparalleled flexibility and ease of use
 
-Numaflow is a Kubernetes-native tool for running massively parallel stream processing. A Numaflow Pipeline is implemented
-as a Kubernetes custom resource and consists of one or more source, data processing, and sink vertices.
+Numaflow, created by the team behind Argo, leverages their expertise to deliver a robust and scalable end-to-end platform for event-driven applications
```
Comment (vigith) [inline README.md:12]:
```suggestion
Numaflow, created by the team at Intuit behind Argo, leverages their expertise to deliver a robust and scalable end-to-end platform for event-driven applications
```

**Review by vigith — COMMENTED** [review]
Shouldn't we also add Serverless?

File: README.md [hunk 2]
```diff
@@ -7,29 +7,27 @@
...
-## Summary
+Welcome to Numaflow, a Kubernetes-native platform designed for developing scalable and reliable event-driven applications. Numaflow decouples event sources and sinks from the processing logic, allowing each component to independently auto-scale based on demand. With out-of-the-box sources and sinks, developers can focus solely on the processing logic without worrying about event consumption, boilerplate code, or operational complexities. Each step of the pipeline can be written in any programming language, offering unparalleled flexibility and ease of use
 
-Numaflow is a Kubernetes-native tool for running massively parallel stream processing. A Numaflow Pipeline is implemented
-as a Kubernetes custom resource and consists of one or more source, data processing, and sink vertices.
+Numaflow, created by the team at Intuit behind Argo, leverages their expertise to deliver a robust and scalable end-to-end platform for event-driven applications
 
-Numaflow installs in a few minutes and is easier and cheaper to use for simple data processing applications than a
-full-featured stream processing platforms.
+![Numaflow Pipeline](./ui/src/images/simple-pipeline.png)
```
Comment (whynowy) [inline README.md:14]: Put the image in `docs/assets/`

File: README.md [hunk 3]
```diff
@@ -7,29 +7,27 @@
 [![Release Version](https://img.shields.io/github/v/release/numaproj/numaflow?label=numaflow)](https://github.com/numaproj/numaflow/releases/latest)
 [![CII Best Practices](https://bestpractices.coreinfrastructure.org/projects/6078/badge)](https://bestpractices.coreinfrastructure.org/projects/6078)
 
-## Summary
+Welcome to Numaflow, a Kubernetes-native platform designed for developing scalable and reliable event-driven applications. Numaflow decouples event sources and sinks from the processing logic, allowing each component to independently auto-scale based on demand. With out-of-the-box sources and sinks, developers can focus solely on the processing logic without worrying about event consumption, boilerplate code, or operational complexities. Each step of the pipeline can be written in any programming language, offering unparalleled flexibility and ease of use
```
Comment (whynowy) [inline README.md:10]: `docs/README.md` also needs to be updated.

Comment (whynowy) [reply README.md:10]: Period.

File: README.md [hunk 4]
```diff
@@ -7,29 +7,27 @@
...
 
-Numaflow is a Kubernetes-native tool for running massively parallel stream processing. A Numaflow Pipeline is implemented
-as a Kubernetes custom resource and consists of one or more source, data processing, and sink vertices.
+Numaflow, created by the team at Intuit behind Argo, leverages their expertise to deliver a robust and scalable end-to-end platform for event-driven applications
```
Comment (whynowy) [inline README.md:12]: Do we really need to add this?

File: README.md (hunk 3 above)
Comment (edlee2121) [inline README.md]:
```suggestion
Welcome to Numaflow! A Kubernetes-native, serverless platform for running scalable and reliable event-driven applications, including in-cluster, realtime, event processing. Numaflow decouples event sources and sinks from the processing logic, allowing each component to independently auto-scale based on demand. With out-of-the-box sources and sinks, and built-in observability, developers can focus on their processing logic without worrying about event consumption, boilerplate code, or operational complexities. Each step of the pipeline can be written in any programming language, offering unparalleled flexibility in using the best programming language for each step, and the ease of using the languages you are most familiar with.
```

File: README.md (hunk 4 above)
Comment (edlee2121) [inline README.md]:
```suggestion
Numaflow, created by the Intuit Argo team in response to community requests for continuous event processing, leverages their expertise to deliver a scalable and robust, serverless platform for event-driven applications.
```

File: README.md [hunk 5]
```diff
@@ -7,29 +7,27 @@
...
+- Inference on streaming data: Perform real-time predictions, e.g., anomaly detection
+- Workflows running in a streaming manner
 
 ## Key Features
 
-- Kubernetes-native: If you know Kubernetes, you already know how to use Numaflow.
-- Language agnostic: Use your favorite programming language.
-- Exactly-Once semantics: No input element is duplicated or lost even as pods are rescheduled or restarted.
-- Auto-scaling with back-pressure: Each vertex automatically scales from zero to whatever is needed.
+- Kubernetes-native: If you know Kubernetes, you already know how to use Numaflow
```
Comment (edlee2121) [inline README.md:25]:
```suggestion
- Kubernetes-native: If you know Kubernetes, you already know how to use Numaflow
- Serverless: Focus on your code and let the system scale up and down based on demand.
```

**Review by edlee2121 — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -7,29 +7,28 @@
...
 
-Numaflow is a Kubernetes-native tool for running massively parallel stream processing. A Numaflow Pipeline is implemented
-as a Kubernetes custom resource and consists of one or more source, data processing, and sink vertices.
+Numaflow, created by the Intuit Argo team to address community needs for continuous event processing, leverages their expertise to deliver a scalable and robust, serverless platform for event-driven applications.
 
-Numaflow installs in a few minutes and is easier and cheaper to use for simple data processing applications than a
-full-featured stream processing platforms.
+![Numaflow Pipeline](./docs/assets/simple-pipeline.png)
```
Comment (vigith) [inline README.md:14]: do we need this? it is an oversimplified diagram.

Comment (syayi) [reply README.md:14]: Updated the image with a better pipeline

**Review by vigith — APPROVED** [review]

**Pull request merged**